### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by EchoAlice**
Changes the name of this branch.  Much more accurate description where I try to tackle issues #1 , #2 , #3 , #4 , #5 within first branch.

May create seperate branches (or PRs?) for seperate RPCs/other functionality later...

### CODE REVIEW

File: src/kbucket.rs [hunk 1]
```diff
@@ -1,146 +1,148 @@
...
+pub enum FindNodeResult {
+    // I don't think this should be "Option<T>".  Fix later
+    Found(Option<Node>),
+    NotFound(Vec<Option<Node>>),
+}
+pub struct SearchResult {
+    pub found: bool,
+    pub bucket_index: usize,
+    pub column_index: usize,
 }
```
Comment (ralexstokes) [inline src/kbucket.rs:21]: you could just make this `Option<NodeRef>` or something where `type NodeRef = (usize, usize)`

not clear who needs these indices outside this module so you may want more expressive typing that just a pair of `usize`...

File: src/kbucket.rs [hunk 2]
```diff
@@ -1,146 +1,148 @@
...
+    ///     - Kademlia Paper
+    ///
+    /// Recieves an id request and returns node information on nodes within
+    /// *its closest bucket* (instead of k-closest nodes) to that id.
+    pub fn find_node(&mut self, id: Identifier) -> FindNodeResult {
+        let result = self.search_table(id);
+        let mut bucket = self.buckets[result.bucket_index];
+
+        if result.found {
+            // Returns Node
```
Comment (ralexstokes) [inline src/kbucket.rs:57]: generally only want to leave comments if they add something beyond what the code is clearly communicating...

File: src/kbucket.rs [hunk 3]
```diff
@@ -1,146 +1,148 @@
...
+                _ => {
                     last_empty_index = i;
                 }
             }
         }
-        // If node wasn't already in bucket -->  return (largest available index, false)
-        println!("Last empty index: {}", last_empty_index);
-        return (false, last_empty_index)
+        SearchResult {
+            found: false,
```
Comment (ralexstokes) [inline src/kbucket.rs:119]: if `found` is helpful info beyond just the binary "was or was not here" then instead I'd suggest an `enum` as something a bit more "idiomatic"

```rust
enum Search {
    Success(usize,usize),
    Failure(usize,usize),
}
```

Comment (EchoAlice) [reply src/kbucket.rs:119]: This seems like the move.  Going to modify code in test branch (sorry) to use this.

Comment (EchoAlice) [reply src/kbucket.rs:119]: Actually might hold off on this... Everything breaks when I move from the SearchResult struct.  I don't think it's worth the headache.  Thoughts?

Comment (EchoAlice) [reply src/kbucket.rs:119]: Implemented here:  https://github.com/EchoAlice/my_kademlia/commit/34e5d5b41a49e860e4e6e54facdafef366381c1e

**Pull request merged**